proposals/0200-raw-string-escaping-revision-1.md

```diff
@@ -0,0 +1,454 @@
+
+```
+// Quoted Text
+let quote = "Alice: "How long is forever?" White Rabbit: "Sometimes, just one second."" 
+let quote = "Alice: \"How long is forever?\" White Rabbit: \"Sometimes, just one second.\""
+
+// and
+
+// Regular Expression
+let ucCaseCheck = "enum\s+.+\{.*case\s+[:upper:]"
+let ucCaseCheck = "enum\\s+.+\\{.*case\\s+[:upper:]"
+```
+
+Escaping hinders readability and interferes with inspection, especially in the latter example, where the content contains secondary escape sequences. Using a raw form ensures the expression can be read and updated as needed in the form that will be passed by the literal string.
+
+### Candidates
+
+A good candidate for using raw strings is non-trivial and is burdened by escaping because it:
+
+* Is obscured by escaping. Escaping actively harms code review and validation.
+* Is already escaped. Escaped material should not be pre-interpreted by the compiler.
+* Requires easy transport between source and code in both directions, whether for testing or just updating source.
+
+The following example is a poor case for using a raw string:
+
+```
+let path = "C:\\AUTOEXEC.BAT"
+```
+
+The example is trivial and the escaping is not burdensome. It's unlikely that the string contents will require any further modification or reuse in a raw form.
+
+### Utility
+
+Raw strings are most valuable for the following scenarios.
+
+**Metaprogramming**: Use cases include code-producing-code. This incorporates utility programming and building test cases without escaping. Apps may generate color scheme type extensions (in Swift, ObjC, for SpriteKit/SceneKit, literals, etc) or date formatters, perform language-specific escaping, create markup, and more.  
+
+Escaping complicates copying and pasting from working code into your source and back. When you're talking about code, and using code, having that code be formatted as an easily updated raw string is especially valuable.
+
+Examples of popular apps that perform these tasks include Kite Compositor and PaintCode. Any utility app that outputs code would benefit in some form.
+
+**Regular expressions**: While regex in general is a much larger problem than raw strings, it is a primary (if not the primary) use case for many Swift developers. Even if we get native regular expressions in Swift, we will still sometimes have to write regular expressions intended for use in other systems using raw literals. 
+
+For instance, if you need to send a regex to a server, or embed it in Javascript, or put it in a SQL query, or (because you're calling existing code which requires it) you need to use `NSRegularExpression` directly, you'll still express a regular expression as a string literal. Raw literals will make that much easier.
+
+Our design's pound syntax might also be useful to extend to native regex literals. If this proposal is accepted, we'd already have it established in the language.
```

Comment:
I think there's a couple of awkward logical leaps and unstated implications in this section. Here's how I would write it instead:

```
**Regular expressions**: While we have bigger plans for regular expressions in the future,
we think they will be a primary use case for raw strings in the short term, and will continue
to have an important place in regex usage in the long term.

Even if we introduce native regular expressions in a future version of Swift, users will still
sometimes have to write regular expressions intended for use in other systems. For instance, if
you need to send a regex to a server, or embed it in Javascript, or put it in a SQL query, or
construct an `NSRegularExpression` and pass it to an existing API which uses that type, you'll
still express that regular expression as a string literal, not a native regex. And when you do,
raw strings will make that much easier.

A raw string feature would thus help with all regular expressions now and some regular
expressions in the future. And if the native regular expression feature involves some form of
quoting and escaping, it can follow the by-then-established precedent of this proposal to
support "raw regexes".
```